warehouse/accounts/views.py

```diff
@@ -259,11 +259,15 @@ def request_password_reset(request, _form_class=RequestPasswordResetForm):
     form = _form_class(request.POST, user_service=user_service)
 
     if request.method == "POST" and form.validate():
-        user = user_service.get_user_by_username(form.username.data)
-        fields = send_password_reset_email(request, user)
-        return {'n_hours': fields['n_hours']}
-
-    return {"form": form}
+        user = user_service.get_user_by_username(form.username_or_email.data)
+        if user is None:
+            user = user_service.get_user_by_email(form.username_or_email.data)
+        if user:
+            fields = send_password_reset_email(request, user)
+
+    token_service = request.find_service(ITokenService, name='password')
+    n_hours = token_service.max_age // 60 // 60
+    return {"n_hours": n_hours}
```

Comment:
These three lines should only happen inside the `if request.method == ...` statement above, we should still `return {"form": form}` at the end of this view.

Reply:
Oh I see! Sorry, I misunderstood your earlier remark on the return value of the view. Making the changes as of now :)